test_bgp_default_originate_timer.py

```diff
@@ -0,0 +1,166 @@
+     address-family ipv4
+      neighbor {{ routers.r2.iface_to('s1').ip4[0].ip }} default-originate route-map default
+     exit-address-family
+    !
+    bgp community-list standard r3 seq 5 permit 65003:1
+    !
+    route-map default permit 10
+     match community r3
+    exit
+    #%   elif router.name == 'r2'
+    router bgp 65002
+     no bgp ebgp-requires-policy
+     neighbor {{ routers.r1.iface_to('s1').ip4[0].ip }} remote-as external
+     neighbor {{ routers.r1.iface_to('s1').ip4[0].ip }} timers 1 3
+     neighbor {{ routers.r1.iface_to('s1').ip4[0].ip }} timers connect 1
+    !
+    #%   elif router.name == 'r3'
+    router bgp 65003
+     no bgp ebgp-requires-policy
+     neighbor {{ routers.r1.iface_to('s2').ip4[0].ip }} remote-as external
+     neighbor {{ routers.r1.iface_to('s2').ip4[0].ip }} timers 1 3
+     neighbor {{ routers.r1.iface_to('s2').ip4[0].ip }} timers connect 1
+     address-family ipv4 unicast
+      redistribute connected route-map r1
+     exit-address-family
+    !
+    route-map r1 permit 10
+     set community 65003:1
+    exit
+    #%   endif
+    #% endblock
+    """
+
+
+class BGPDefaultOriginateTimer(TestBase, AutoFixture, topo=topology, configs=Configs):
+    @topotatofunc
+    def bgp_default_received_from_r1(self, _, r1, r2):
+        expected = {
+            "paths": [
+                {
+                    "nexthops": [
+                        {
+                            "hostname": "r1",
+                            "ip": str(r1.iface_to("s1").ip4[0].ip),
+                        }
+                    ],
+                }
+            ],
+        }
+
+        yield from AssertVtysh.make(
+            r2,
+            "bgpd",
+            f"show bgp ipv4 unicast 0.0.0.0/0 json",
+            maxwait=10.0,
+            compare=expected,
+        )
+
+    @topotatofunc
+    def bgp_default_received_from_r1_2(self, _, r1, r2, r3):
+        expected = {
+            "paths": [
+                {
+                    "nexthops": [
+                        {
+                            "hostname": "r1",
+                            "ip": str(r1.iface_to("s1").ip4[0].ip),
+                        }
+                    ],
+                }
+            ],
+        }
+
+        yield from AssertVtysh.make(
+            r1,
+            "vtysh",
+            """
+            configure terminal
+                router bgp
+                    bgp default-originate timer 10
+            """,
+            compare="",
+        )
+
+        yield from AssertVtysh.make(
+            r3,
+            "vtysh",
+            """
+            configure terminal
+                route-map r1 permit 10
+                    set metric 1
+            """,
+            compare="",
+        )
+
+        yield from AssertVtysh.make(
+            r2,
+            "bgpd",
+            f"show bgp ipv4 unicast 0.0.0.0/0 json",
+            maxwait=10.0,
```

Comment:
… so this wait is not long enough for the 10s above.

⇒ shorten the 10s above, and make this down here maybe 15s (there is some additional delay, probably from internal BGP timers) ⇒ then it should work